### ISSUE TITLE
Capture (tethering) module improvment to greatly improve the studio work efficiency: Auto-process images

### ISSUE THREAD

**Issue by nbremond77** [darktable-org/darktable#2930]
Hello, I'd like to propose 3 simple and very useful improvements to the capture tethering module:

**Use case:**
- In studio shooting session, with a defined lighting setup and camera settings, I take many pictures that will need (almost) the same post processing, at least for all the basics (exposure, color adjustment/BW conversion, ...). It would be wonderful if DT could theter (capture) the images from the camera in (almost) real time and automatically apply a defined post processing on each picture.

- In studio shooting, I also need to monitor in almost "real time" any clipped area in the capture's module window, in order to check the good exposure of the captured images.

- The second need is a folder monitoring and auto-download capability for the capture/tethering module: My camera (5D mark IV) is able to FTP every image to a server. I'd like DT's capture module to be able to monitor the folder of this FTP server, and automatically import any new image, and automatically apply the pre-defined processing as described above.

**Proposed solution:**
- Add an option to the capture module to automatically apply the processing (as describe in the xml sidecar file) of the previously captured image to any new captured image, and display the resulting, post-processed, image. This way we can very efficiently monitor the quality of the shoots, almost in real time, and show almost "final" pictures to the model, without going back and forth to the darkroom module.

- BTW, we also need an option to show the clipped areas of the image (burned-too bright or pitch black areas) as we have in the darkroom module. This is very useful to control that the captured image has a good exposure.

- Add an option to monitor the content of a defined folder, and automatically import any new image from this folder, the same way as if this image was tethered from the camera. This will be a great backup plan when we have issues with the tethering mode. In addition, using the FTP transfer mode of my camera keep storing the image in the compact flash of the camera, while in tethering mode the images are not stored locally in the camera, whit doesn't seem very safe...

**Alternatives:**
Instead of applying the last captured image processing, we could think of applying a defined style (as defined in the main DT window), but I think this will be less ergonomic and less efficient because we would need to save a new style to take into account any changes in the post processing.

**Benefits:**
- Greatly increase the efficiency of the workflow in studio work,
- Getting closer to the very efficient workflow pro photographers like Peter Coulson uses in studio with the Phocus software (Hasselblade) : https://www.youtube.com/watch?v=3Nk3CBBgzCw    or   https://www.youtube.com/watch?v=K7oFGanFiXk
- Be able to detect any clipping of the captured images in studio work
- Be able to show the model, images that are closed to the final, post processed, images, in black and white for example...

**Thanks a lot !!!**

And I want to thanks the whole team for this wonderful and very good Darktable software I use several times a week with a lot of pleasure and very good results. And I like very much the open source philosophy behind it.
Best regards,
Bernard.

**Comment by dim162** [comment]
I don't think it's a good idea to develop the tethering in Dt... It would be as good as Qdslrdashboard which do the job very well. Maybe a collaboration between the 2 softwares  can be imagined?!

**Comment by nbremond77** [comment]
Hello,
I checked Qdslrdashboard and as far as I can see, it doesn't meet the requirements I explained above.

My main point is to (add the capability to) automatically apply the last development parameters to every captured images. This will bring the capability to see a (close to) final images in real-time with the model during studio shooting sessions. This will also facilitate fine tuning of the settings and lighting during the studio session.

This will also bring DT to the next level for studio shootings, as the Hasselblad Phocus software does...

**Comment by matrixik** [comment]
Seeing more videos from Peter Coulson when he is using Phocus I would really love to have similar functionality in any open source that work with more camera brands (I want it for my Sony A7III).

So I would prefer this issue not to be closed.

**Comment by nbremond77** [comment]
Hello,
Thanks. 
I fully agree. Phocus is really  the perfect app for studio work... unfortunately it works only with hasselblad... which is far out of y budget.

I think DT can provide pretty good studio work functionalities with reasonable effort:

- Add the capability to automatically apply a pre-defined style, or the xml development file of the last picture, or a selected picture) to every thetered pictures: Then the thetered images with have a look close to the final postprocessed image (black and white, color corrected...). This allows the model and the photograph to check the shoots almost in realtime (see Peter Coulson videos)

- Add the overexposure and underexposure indicators (blinking areas) in the camera view (thetering mode). This allow the check in realtime if the exposure is correct.

- In thetering mode, in addition to the capture in DT, add the capability to have the picture also saved in the camera sd-card or cf-flash. This is important to keep a backup in the camera if something goes wrong during the thetering.

Please could you keep this topic open and add these features to a next DT release ? 
Thanks in advance.
Best regards.

**Comment by johnny-bit** [comment]
While this would require more devs with tethering-able cameras and some deep gphoto2 work, I see no downsides to having more powerful tethering :)

**Comment by TurboGit** [comment]
@johnny-bit : I fully agree having a single integrated tool to work on is certainly better that having to handle different tools with different ergonomic and ways to add tags, classify, etc... So a decent tethering mode is dt is really something we want to keep.

**Comment by matrixik** [comment]
Yes, it would be great but from what I read on discuss.pixls.us and in some github issues tethering have now almost no one working on it. It's not rewritten like rest of DT so it's using stuff that was removed from other places so it's getting more broken over time. And it was almost removed completely from DT once, it survive only because one person said that he is using it in studio.

**Comment by matrixik** [comment]
Looks like @github-actions bot got broken on this issue. It's still did not remove `no-issue-activity` label. Does that mean that it will still auto close it on Jan 30 2021?

**Comment by johnny-bit** [comment]
re-bot: no-issue-activity needs to be taken off manually. and time seems to count from last activity and not from date of no-issue-acitvity tag

**Comment by Rincewind15** [comment]
Is there a way to revive this request?
It would enhance the usability of darktable and would put it next to Capture One or Phocus.

**Comment by nbremond77** [comment]
Yes I fully agree ! Please could you revive this proposal as it will be a great, and long time expected, addition to DT !!!
Thanks in advance.

As the tethering capability is already in place in DT... only few nuts and bolds are missing to have a great tool I think... and these nuts and bolds are mostly already implemented in other parts of DT it seems:

- clipping alert in the tethering vue, 
- Full screen preview (and second screen preview), with the optional clipping indicator,  that can stay active during the shooting
- on the fly, auto-applying style or .xml adjustments file of a selected already existing image, on all tethered images
- and optionally, the capability to have the tethered images saved on both the camera SD card AND on DT - just to be safe in case of tethering problem
- and the capability to reset the tethering (it crashed from times to times) without having to fully restart DT and the camera

Thanks in advance to take this into consideratio... Maybe in the next DT summer of code ;-)

**Comment by nbremond77** [comment]
Thanks for your feedback. 
Just wondering, could those features be implemented  as LUA plugin(s) ? Or do one need to write DT native C++ code ?

**Comment by wpferguson** [comment]
Some things can be done from Lua, such as creating and applying styles.

**Comment by nbremond77** [comment]
Is there any Lua trigger to automatically apply a style (or a .xml cart file) on every captured (tethered) image as soon as they are received by DT ?

**Comment by wpferguson** [comment]
If the image loads into darkroom, then there is a darkroom-image-loaded event that you could catch

**Comment by nbremond77** [comment]
OK. Thanks

**Comment by wpferguson** [comment]
The API documentation is at https://docs.darktable.org/lua/stable/lua.api.manual/

**Comment by wpferguson** [comment]
I still think this is valid.  I can implement some parts using Lua.

**Comment by wpferguson** [comment]
I've got the folder watch and copy implemented.  Applying a style on load works, I use it all the time in my processing.  Showing over/under exposed areas is a toggle shortcut or can be automated.
